File: solutions/java/log-levels/1/src/main/java/LogLevels.java
public class LogLevels {
    
    public static String message(String logLine) {
        return logLine.stripTrailing().replaceFirst("\\[.*\\]: ", "").stripLeading();
    }

    public static String logLevel(String logLine) {
        return logLine.replaceFirst("\\[(.*)\\]: .*", "$1").toLowerCase();
    }

    public static String reformat(String logLine) {
        return message(logLine) + " (" + logLevel(logLine).strip() + ")";
    }
}

File: solutions/fsharp/log-levels/1/LogLevels.fs
module LogLevels

let message(logLine: string): string =
    logLine
    |> _.Split(": ", 2)
    |> Array.last
    |> _.Trim()

let logLevel(logLine: string): string =
    let level = logLine.Split(": ", 2)[0]
    level.Substring(1, String.length level - 2).ToLower()

let reformat(logLine: string): string = sprintf "%s (%s)" (message logLine) (logLevel logLine)
